source/Octopus.Client.Tests/Operations/RegisterKubernetesClusterOperationFixture.cs

```diff
@@ -101,5 +108,147 @@ public async Task ShouldCreateNewPollingKubernetesTentacle()
 
             machineResources.Should().ContainSingle().Which.Endpoint.Should().BeOfType<KubernetesTentacleEndpointResource>().Which.Should().BeEquivalentTo(new KubernetesTentacleEndpointResource(new PollingTentacleEndpointConfigurationResource("ABCDEF", "poll://ckyhfyfkcbmzjl8sfgch/")));
         }
+
+        [Test]
+        public async Task ShouldOnlyUpdateEndpointConnectionConfigurationOnExistingMachine()
+        {
+            var updatedMachines = new List<MachineResource>();
+            await client.Update("/machines/whatever/1", Arg.Do<MachineResource>(m => updatedMachines.Add(m)), Arg.Any<object>(), Arg.Any<CancellationToken>());
+
+            environments.Items.Add(new EnvironmentResource {Id = "environments-1", Name = "UAT", Links = LinkCollection.Self("/api/environments/environments-1").Add("Machines", "/api/environments/environments-1/machines")});
+            environments.Items.Add(new EnvironmentResource {Id = "environments-2", Name = "Production", Links = LinkCollection.Self("/api/environments/environments-2").Add("Machines", "/api/environments/environments-2/machines")});
+
+            machines.Items.Add(new MachineResource
+            {
+                Id = "machines/84", EnvironmentIds = new ReferenceCollection(new[] { "environments-1" }), Name = "Mymachine", Links = LinkCollection.Self("/machines/whatever/1"),
+                Endpoint = new KubernetesTentacleEndpointResource(new ListeningTentacleEndpointConfigurationResource("123456", "myMachine.test.com") { ProxyId = "proxy-2" })
+            });
+
+            proxies.Items.Add(new ProxyResource { Id = "proxy-1", Name = "MyNewProxy", Links = LinkCollection.Self("/api/proxies/proxy-1") });
+
+            operation.MachineName = "Mymachine";
+            operation.TentaclePort = 10930;
+            operation.TentacleThumbprint = "ABCDEF";
+            operation.TentacleHostname = "my-new-machine.test.com";
+            operation.CommunicationStyle = CommunicationStyle.TentaclePassive;
+            operation.EnvironmentNames = new[] {"Production"};
+            operation.ProxyName = "MyNewProxy";
+
+            await operation.ExecuteAsync(serverEndpoint).ConfigureAwait(false);
+
+            var thing = updatedMachines.Should().ContainSingle().Which;
+
+            thing.Should().BeEquivalentTo(new MachineResource
+            {
+                Id = "machines/84",
+                Name = "Mymachine",
+                EnvironmentIds = new ReferenceCollection("environments-1"),
+                Endpoint = new KubernetesTentacleEndpointResource(new ListeningTentacleEndpointConfigurationResource("ABCDEF", "https://my-new-machine.test.com:10930/") {ProxyId = "proxy-1"}),
+                Links = LinkCollection.Self("/machines/whatever/1")
+            }, cfg => cfg.RespectingRuntimeTypes());
+        }
+
+        [Test]
+        public async Task ShouldUpdateExistingMachineWhenForceIsEnabled()
```

Comment:
I ended up adding all these tests from the `RegisterMachineOperationFixture` just to ensure the functionality is maintained for `RegisterKubernetesClusterOperation`, even if someone wants to change `RegisterMachineOperation`.